class.geocoder.php

```diff
@@ -167,10 +167,15 @@ private function osm_geocode ( $address ) {
         $json = $this->get_url($geocode_url);
         $json = json_decode($json);
         
-        return (Object) array(
-            'lat' => $json[0]->lat,
-            'lng' => $json[0]->lon,
-        );
+        if (isset($json[0]->lat) && isset($json[0]->lon)) {
+            return (Object) array(
+                'lat' => $json[0]->lat,
+                'lng' => $json[0]->lon,
+            );
+        } else {
+            throw new Exception('No Address Found.');
+            return false;
```

Comment:
I doubt the return does anything.  I'm also a bit nervous about throwing an Exception

Reply:
I guess the google geocoder does the same.  Wonder if I should add some try/catches around these

Reply:
Or should it return default lat and lng?

Reply:
Nah.  I think throwing is fine here, since that's what the other geocoder is doing. I'm just realizing we're not catching those exceptions anywhere